src/Gui/Dialogs/DlgExpressionInput.cpp

```diff
@@ -644,25 +766,49 @@ void DlgExpressionInput::onCheckVarSets(int state) {
     }
     else {
         okBtn->setEnabled(true); // normal expression
+        adjustSize();
     }
 }
 
-void DlgExpressionInput::onVarSetSelected(int)
+void DlgExpressionInput::preselectGroup()
 {
-    QTreeWidgetItem* selected = treeWidget->currentItem();
+    auto paramExpressionEditor = App::GetApplication().GetParameterGroupByPath(
+            "User parameter:BaseApp/Preferences/ExpressionEditor");
+    std::string lastGroup = paramExpressionEditor->GetASCII("LastGroup", "");
 
-    if (selected) {
-        // if group is known, fill it in
-        QVariant variantGroup = selected->data(0, ROLE_GROUP);
-        if (variantGroup.isValid()) {
-            ui->lineEditGroup->setText(variantGroup.toString());
-        }
-        else {
-            ui->lineEditGroup->clear();
-        }
+    if (lastGroup.empty()) {
+        return;
+    }
+
+    int index = comboBoxGroup.findText(QString::fromStdString(lastGroup));
+    if (index == -1) {
+        return;
+    }
+
+    comboBoxGroup.setCurrentIndex(index);
```

Comment:
Total nit: that I'd probably write reversed:


```suggestion
    if (int index = comboBoxGroup.findText(QString::fromStdString(lastGroup)); index == -1) {
        comboBoxGroup.setCurrentIndex(index)
    }
```